cli/src/pcluster/config/update_policy.py

```diff
@@ -125,6 +134,66 @@ def is_slurm_queues_change(change):
     return any(path.startswith("SlurmQueues[") for path in change.path)
 
 
+def extract_type_and_name_from_path(path):
+    obj_type = re.search(".+(?=\\[)", path).group(0)
+    obj_name = re.search("(?<=\\[).+(?=\\])", path).group(0)
+    return obj_type, obj_name
+
+
+def get_q_from_config(change, config):
+    q_type, q_name = None, None
+    for path in change.path:
+        if re.search("Queues\\[", path):
+            q_type, q_name = extract_type_and_name_from_path(path)
+    if q_type and q_name:
+        for queue in config.get("Scheduling", {}).get(q_type, {}):
+            if queue.get("Name", None) == q_name:
+                return queue
+    return None
+
+
+def get_cr_from_config(change, config):
+    cr_type, cr_name = None, None
+    for path in change.path:
+        if re.search("ComputeResources\\[", path):
+            cr_type, cr_name = extract_type_and_name_from_path(path)
+    if cr_type and cr_name:
+        queue = get_q_from_config(change, config)
+        for compute_resource in queue.get(cr_type, {}):
+            if compute_resource.get("Name", None) == cr_name:
+                return compute_resource
+    return None
+
+
+def is_placement_group_managed_for_compute_resource(queue_networking, compute_resource_networking):
+    cr_pg = compute_resource_networking.get("PlacementGroup", None)
+    cr_pg_enabled = compute_resource_networking.get("PlacementGroup", {}).get("Enabled")
+    q_pg_enabled = queue_networking.get("PlacementGroup", {}).get("Enabled")
+    if (cr_pg is None and q_pg_enabled) or cr_pg_enabled:
+        return True
+    return False
+
+
+def is_managed_placement_group_deletion(change, patch):
```

Comment:
Does the logic work for the following config changes:
Base config:
```
Queues:
  - PlacementGroup: Enabled
    ComputeResources: <No placement group specification>
```
target config:
```
Queues:
  - <No placement group specification>
    ComputeResources: <Every compute resource has PlacementGroup: Enabled>
```

We should expect no errors generated from the change above, right?

Reply:
the update process will evaluate changes at the queue level separately from the compute resource.  If we try to evaluate all the possible combinations we will end up with tons of redundant update checks.  So in this case for example, we would have a change for the queue level pg as well as a change for the compute resource level pg's.  We would then be checking every compute resource whenever there is a change to the queue level.  It seems like a big coupling to be using the data from a different change.  So I decided to separate the checks and if the user removes the managed placement group from the queue level regardless of what they do at the compute resource level, they will have to stop the fleet, even if they compensate at the compute resource level.

Reply:
Leave for now with a decision doc to follow for future improvement